### PR TITLE
Convert Velero To Helm Chart Implementation

### PR DESCRIPTION
**Pull request by shaneutt**
This PR converts Velero to a chart based implementation and utilizes [extrasteps](https://github.com/kubeaddons/extrasteps) to deploy a non-production fallback Minio cluster for the Velero Addon if no other backend is provided.

### CODE REVIEW

File: templates/velero.yaml
Comment (shaneutt) [inline templates/velero.yaml]:
```suggestion
        # initialize-velero manages deploying a fallback backend (e.g. Minio) for Velero if needed,
        # and also will manage the credentials and secrets for that backend and Velero automatically.
        - name: initialize-velero
```

**Comment by shaneutt** [comment]
squashed

File: templates/velero.yaml
Comment (gracedo) [inline templates/velero.yaml]: Can we enable metrics exporting? There is a `metrics.enabled` config field that is defaulted to `false`

Comment (shaneutt) [reply templates/velero.yaml]: Sure, sounds good :+1:

Comment (shaneutt) [reply templates/velero.yaml]: feb57b8bc82d7b8007d448069af3e897cb3c10f6 takes care of enabling metrics, and enabling the serviceMonitor for the Prometheus operator. Let me know if there's anything else you'd like tweaked before we merge @gracedo :+1:

File: templates/velero.yaml
Comment (hectorj2f) [inline templates/velero.yaml]: Please, let's simply add a link to the Velero helm, this is making this YAML hard to read and maintain :/.

**Review by hectorj2f — CHANGES_REQUESTED** [review]
Let's remove all that commented code in the yaml please 🙏 

Question: What is the size of the minio volume ?

File: templates/velero.yaml
Comment (shaneutt) [reply templates/velero.yaml]: > Let's remove all that commented code in the yaml please

Why?

> Question: What is the size of the minio volume?

[10g](https://github.com/kubeaddons/extrasteps/blob/master/pkg/minio/minio.go#L164) per instance (default 4 instances)

> Please, let's simply add a link to the Velero helm, this is making this YAML hard to read and maintain :/.

I'm confused by this assertion. Can you help me to better understand your take on this please?

**Review by gracedo — DISMISSED** [review]

File: templates/velero.yaml
Comment (s12chung) [reply templates/velero.yaml]: i'd prefer a link to, but it's a non-blocker.

**Review by s12chung — DISMISSED** [review]

File: templates/velero.yaml
Comment (hectorj2f) [reply templates/velero.yaml]: What is the value of adding all these text ? In my opinion, the value could be provided by a link to the velero helm and minio operator. There you can add the README and additional explanations about this addon. Let's keep this file as short as possible.

YOu could add this text to the mesosphere/chart of velero.

Comment (hectorj2f) [reply templates/velero.yaml]: Otherwise we could add a docs section here for velero. However I think that a link would be helpful enough.

Comment (shaneutt) [reply templates/velero.yaml]: I must respectfully disagree with your position on this, however since this is a blocker and holding up other work I've made a compromise and removed the testing section which included hints about how to deploy the addon:

a25e1eaa83c79961b87dacc5d6b1eaaa48b0e82b

I've also created a JIRA which covers the need to testing tooling for addons such as this was attempting to provide a little bit of for velero:

[DCOS-56456](https://jira.mesosphere.com/browse/DCOS-56456)

Let me know if this is an acceptable compromise from your end. I will mark this as resolved, please mark it as unresolved again if there is more you feel needs to be done.

**Review by gracedo — APPROVED** [review]

File: templates/velero.yaml
Comment (s12chung) [reply templates/velero.yaml]: I'm good if hector is ok. I still prefer a link.

Comment (shaneutt) [reply templates/velero.yaml]: What link @s12chung? There are three links below did you want something else added or did you mean JUST a link?

**Review by hectorj2f — APPROVED** [review]

**Pull request merged**
